nikola/plugins/command/check.py

```diff
@@ -44,33 +46,26 @@
 from nikola.utils import get_logger, req_missing
 
 
-def _call_nikola_list(l, site, arguments):
-    class NotReallyAStream(object):
-        """A massive hack."""
-        out = []
+def _call_nikola_list(site):
+    files = []
+    deps = defaultdict(list)
+    for task in generate_tasks('render_site', site.gen_tasks('render_site', "Task", '')):
```

Comment:
You can just talk to `self.site.doit.task_loader`.  You also can save the generated tasks there, if any.